### PR TITLE
Smolagents announcement post

### PR DESCRIPTION
**Pull request by aymeric-roucher**

### CODE REVIEW

File: smolagents.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
+---
+title: "Introducing smolagents: simple agents that write actions in code."
+thumbnail: /blog/assets/beating-gaia/thumbnail.jpeg
+authors:
+  - user: m-ric
+  - user: merve
+---
+# Introducing *smolagents*, a simple library to build agents
```
Comment (merveenoyan) [inline smolagents.md:8]: with single asterisk it's italic, we need to put two to make it bold I think

File: smolagents.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+agent = CodeAgent(tools=[DuckDuckGoSearchTool()], model=HfApiModel())
+
+agent.run("How many seconds would it take for a leopard at full speed to run through Pont des Arts?")
+```
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/smolagents/smolagents.gif" />
+</div>
+
+### 🤔 What are agents?
```
Comment (merveenoyan) [inline smolagents.md:24]: perhaps we can put a nice ToC before, so this needs to be '##'

File: smolagents.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+1. Want some knowledge on the trips? ⇒ give them access to a search bar to search your knowledge base
+2. Wants to talk to sales? ⇒ let them type in a contact form.
+
+If that deterministic workflow fits all queries, by all means just code everything! This will give you a 100% reliable system with no risk of error introduced by letting unpredictable LLMs meddle in your workflow. For the sake of simplicity and robustness, it's advised to regularize towards not using any agentic behaviour. 
+
+But what if the workflow can't be determined that well in advance? 
+
+For instance, a user wants to ask : `"I can come on Monday, but I forgot my passport so risk being delayed to Wednesday, is it possible to take me and my stuff to surf on Tuesday morning, with a cancellation insurance?"` This question hinges on many factors, and probably none of the predetermined criteria above will suffice for this request.
+
+If the pre-determined workflow falls short too often, that means you need more flexibility,.
```
Comment (merveenoyan) [inline smolagents.md:77]:
```suggestion
If the pre-determined workflow falls short too often, that means you need more flexibility.
```

File: smolagents.md (hunk 2 above)
Comment (aymeric-roucher) [reply smolagents.md:24]: Great idea! It's added

**Comment by merveenoyan** [comment]
we should add the plot, link the models and thumbnail and we can launch

**Review by merveenoyan — APPROVED** [review]
checked on new-blog, LGTM

**Pull request merged**